server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java

```diff
@@ -88,48 +97,75 @@ protected void startTransport() {
       transport = new NettyTransport(address, configuration, getQualifiedName(), cacheManager);
       transport.initializeHandler(getInitializer());
 
-      // Register transport MBean regardless
-      registerTransportMBean();
+      // Register transport and worker MBeans regardless
+      registerServerMBeans();
 
       try {
          transport.start();
       } catch (Throwable re) {
          try {
-            unregisterTransportMBean();
+            unregisterServerMBeans();
          } catch (Exception e) {
             throw new CacheException(e);
          }
          throw re;
       }
    }
 
-   protected void registerTransportMBean() {
+   protected ThreadPoolExecutor getExecutor() {
+      if (this.executor == null || this.executor.isShutdown()) {
+         DefaultThreadFactory factory = new DefaultThreadFactory(getQualifiedName() + "-ServerHandler");
+         int workerThreads = getWorkerThreads();
+         this.executor = new ThreadPoolExecutor(
+               workerThreads,
+               workerThreads,
+               0L, TimeUnit.MILLISECONDS,
+               new LinkedBlockingQueue<>(),
```

Comment:
Not sure about having an unlimited queue here.

Reply:
That's how it has always been. The intent of this PR is not to change that behaviour atm.